### PR TITLE
Added a WaitForPageLoad static method to the Scenario class, and call…

### PR DESCRIPTION
**Pull request by jewilder**
…s by the 12 scenarios that make up the "representative" workload.  This ensures that the page is fully loaded before trying to interact with elements that otherwise may not exist yet.

yahooNews - added support for both of the alternative layouts known at this time.
tumblr - sometimes tumblr doesn't load content.  Added loop that reloads the page until content exists, up to 5 times.

### CODE REVIEW

**Comment by brheenan** [comment]
Hey, it looks good. Two things:

1. Minor and easy to fix: This PR is a breaking change since running this will yield potentially different results than prior to these changes (since we're adding some logic to wait after the page loads). Because we're introducing a breaking change, we need to rev the version. Looks like you'd be updating from 4.1 to 5.0. See this commit for an example on how to do this: 844dc3dbbf0a18676ad8c91261d2bfd163615fad
2. I'm concerned about that implementation of Tumblr. Putting in extra retries within a single iteration and allowing that iteration to continue will introduce more variability. If a page fails and requires a reload, the whole iteration should be thrown out. If Tumblr is causing issues consistently, I think it would be better to remove/replace it rather than wrapping it in its own special retry logic

File: BrowserEfficiencyTest/Scenario.cs [hunk 1]
```diff
@@ -40,5 +43,12 @@ internal abstract class Scenario
 
         // Override this function with the "stuff" to do in the scenario
         public abstract void Run(RemoteWebDriver driver, string browser, CredentialManager credentialManager);
+
+        public static void WaitForPageLoad(RemoteWebDriver driver, int timeoutSec = 30)
```
Comment (TychosNose) [inline BrowserEfficiencyTest/Scenario.cs]: This method is an extension of the RemoteWebDriver class. We place all RemoteWebDriver extensions in the RemoteWebDriverExtension.cs file. Please move this method there and add 'this' to make the definition be:
public static void WaitForPageLoad(this RemoteWebDriver driver, int timeoutSec = 30)
Then update all the calls in the rest of the scenarios to be driver.WaitForPageLoad();

**Comment by jewilder** [comment]
I pushed commits that I believe covers everything discussed.  Please take a look.

**Comment by TychosNose** [comment]
Looks good to me. Brandon, do you have any concerns?

**Pull request merged**